internal/cmd/api-key/command_delete.go

```diff
@@ -28,17 +25,42 @@ func (c *command) newDeleteCommand() *cobra.Command {
 
 func (c *command) delete(cmd *cobra.Command, args []string) error {
 	c.setKeyStoreIfNil()
-	apiKey := args[0]
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmYesNoMsg, resource.ApiKey, apiKey)
-	if ok, err := form.ConfirmDeletion(cmd, promptMsg, ""); err != nil || !ok {
+	if confirm, err := c.confirmDeletion(cmd, args); err != nil {
 		return err
+	} else if !confirm {
+		return nil
+	}
+
+	deleteFunc := func(id string) error {
+		if r, err := c.V2Client.DeleteApiKey(id); err != nil {
+			return errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, r)
```

Comment:
Don't we only care about the existence of an error here? Why are we wrapping it?

Reply:
This is the func for deleting, where we do want the error. The one for describing (existence check) is where we don't need the error.